vault/resource_identity_oidc_key_test.go

```diff
@@ -0,0 +1,217 @@
+	key := acctest.RandomWithPrefix("test-key")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testProviders,
+		CheckDestroy: testAccCheckIdentityOidcKeyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccIdentityOidcKeyConfig(key),
+				Check: resource.ComposeTestCheckFunc(
+					testAccIdentityOidcKeyCheckAttrs(),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "name", key),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "rotation_period", "86400"),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "verification_ttl", "86400"),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "algorithm", "RS256"),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "allowed_client_ids.#", "0"),
+				),
+			},
+			{
+				ResourceName:      "vault_identity_oidc_key.key",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccIdentityOidcKeyUpdate(t *testing.T) {
+	key := acctest.RandomWithPrefix("test-key")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testProviders,
+		CheckDestroy: testAccCheckIdentityOidcKeyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccIdentityOidcKeyConfig(key),
+				Check:  testAccIdentityOidcKeyCheckAttrs(),
+			},
+			{
+				Config: testAccIdentityOidcKeyConfigUpdate(key),
+				Check: resource.ComposeTestCheckFunc(
+					testAccIdentityOidcKeyCheckAttrs(),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "name", key),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "rotation_period", "3600"),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "verification_ttl", "3600"),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "algorithm", "ES256"),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "allowed_client_ids.#", "1"),
+				),
+			},
+			{
+				Config: testAccIdentityOidcKeyConfig(key),
+				Check: resource.ComposeTestCheckFunc(
+					testAccIdentityOidcKeyCheckAttrs(),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "name", key),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "rotation_period", "86400"),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "verification_ttl", "86400"),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "algorithm", "RS256"),
+					resource.TestCheckResourceAttr("vault_identity_oidc_key.key", "allowed_client_ids.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIdentityOidcKeyDestroy(s *terraform.State) error {
+	client := testProvider.Meta().(*api.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "vault_identity_oidc_key" {
+			continue
+		}
+		secret, err := client.Logical().Read(identityEntityIDPath(rs.Primary.ID))
```

Comment:
Do you instead want something like: `resp, err := identityOidcKeyApiRead(rs.Primary.Attributes["name"], client)` ?

Not that when looking at this I discovered that Vault is incorrectly returning a 400 for key not found instead of 404. I've just fixed that, but in the interim you may need the subsequent check to be something like:

`if err != nil && !strings.Contains(err.Error(), "no named key found") {`

Reply:
I've fixed this and added the check in `identityOidcKeyApiRead`(commit 34517b60662fbe3e3e560cfa3382f00f19da7442).